internal/service/cloudbackupschedule/resource_cloud_backup_schedule.go

```diff
@@ -674,3 +740,79 @@ func policyItemID(policyState map[string]any) *string {
 	}
 	return nil
 }
+
+func shouldUseOldAPI(d *schema.ResourceData, isCreate bool) (bool, error) {
+	copySettings := d.Get("copy_settings")
+	if isCopySettingsNonEmptyOrChanged(d) {
+		return CheckCopySettingsToUseOldAPI(copySettings.([]any), isCreate)
+	}
+	return false, nil
+}
+
+func isCopySettingsNonEmptyOrChanged(d *schema.ResourceData) bool {
+	copySettings := d.Get("copy_settings")
+	return copySettings != nil && (conversion.HasElementsSliceOrMap(copySettings) || d.HasChange("copy_settings"))
+}
+
+// CheckCopySettingsToUseOldAPI verifies that all elements in tfList use either `replication_spec_id` or `zone_id`
+// Returns an error if any element has both `replication_spec_id` and `zone_id` set during create
+// and returns a bool if the old API should be used or not
+func CheckCopySettingsToUseOldAPI(tfList []any, isCreate bool) (bool, error) {
```

Comment:
should we validate that if zone_id is used, it is used in all copy settings? Not certain for the best place for adding this validation

Reply:
I decided not to do that because as per current logic, we ensure that old API is called only if ALL elements have rep_spec_id set, otherwise it will always default to calling the new API. In this case (if elements contain missing zone_id or mixed zone_id & rep_spec_id), then we let the new API handle that by returning an error. 
Let me know WYT, happy to discuss more on this.

Reply:
I see. Would make sure we call this out in the docs and migration guide: when switching to using zone_id, all copy_settings must be adjusted in the same update plan.

Reply:
captured this in the doc ticket for this https://jira.mongodb.org/browse/CLOUDP-262919